plugins/com.google.cloud.tools.eclipse.appengine.validation.test/src/com/google/cloud/tools/eclipse/appengine/validation/WebXmlScannerTest.java

```diff
@@ -43,25 +47,34 @@
 public class WebXmlScannerTest {
 
   private static IFile resource;
-  private static IProject project;
-  private WebXmlScanner scanner = new WebXmlScanner();
+  private static IFile servletClass;
+  private static IJavaProject javaProject;
+  private WebXmlScanner scanner;
   
-  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator();
+  @ClassRule public static TestProjectCreator projectCreator = new TestProjectCreator().withClasspathContainerPath("src/main/java");
   
   @BeforeClass
   public static void setUpBeforeClass() throws CoreException {
-    project = projectCreator.getProject();
+    IProject project = projectCreator.getProject();
+    javaProject = JavaCore.create(project);
+    
+    IPackageFragmentRoot root = javaProject.getPackageFragmentRoot("src/main/java");
+    root.createPackageFragment("com.test", false, null);
```

Comment:
I'm having trouble programmatically adding a package to an IProject in WebXmlScannerTest. This line throws `Java Model Exception: Java Model Status [java is read-only]`

Tried adding attributes to folders in createFolders() so they weren't read-only, but no luck.